SCF-Session-7-Assignment/src/question1/SparseMatrix.java

```diff
@@ -0,0 +1,258 @@
+	 */
+	public int[][] transposeOfSparseMatrix()
+	{
+		assert getSparseMatrix().length == 0 : "Matrix size is zero";
+		
+		int[][] matrix = convertSparseToTripletMatrix(this.getSparseMatrix());
+		int[][] transposeMatrix = new int[matrix.length][3];
+		
+		
+		for(int i = 0; i < matrix.length; i++)
+		{
+			transposeMatrix[i][0] = matrix[i][1];
+			transposeMatrix[i][1] = matrix[i][0];
+			transposeMatrix[i][2] = matrix[i][2];
+		}
+		
+		return convertTripletToSparseMatrix(transposeMatrix, numberOfColumn, numberOfRow);
+	}
+	
+	/**
+	 * This method checks whether the sparse matrix is symmetric or not
+	 * @return {boolean}
+	 */
+	public boolean isSparseMatrixSymmetric()
+	{
+		assert numberOfRow != numberOfColumn : "Only square matrix can be symmetric";
+		
+		int[][] sparseTransposeMatrix = transposeOfSparseMatrix();
+		
+		if(Arrays.deepEquals(this.getSparseMatrix(), sparseTransposeMatrix))
+			return true;
+		else
+			return false;
+	}
+	
+	/**
+	 * This method adds two sparse matrices
+	 * @param inputMatrix is the matrix whose addition is to be carried out with the current matrix
+	 * @return {int[][]} addition of the given two matrices
+	 */
+	public int[][] addTwoSparseMatrices(int[][] inputMatrix)
+	{
+		int[][] tripletMatrix1 = convertSparseToTripletMatrix(getSparseMatrix());
+		int[][] tripletMatrix2 = convertSparseToTripletMatrix(inputMatrix);
+		int[][] resultantMatrix = new int[tripletMatrix1.length + tripletMatrix2.length][3];
+		
+		assert tripletMatrix1.length == 0 || tripletMatrix2.length == 0 : "Matrices cannot be added";
+		
+		if(numberOfRow == inputMatrix.length && numberOfColumn == inputMatrix[0].length)
+		{
+			int rowOfMatrix1;
+			int columnOfMatrix1;
+			int i;
+			int resultantMatrixRow = 0;
+			
+			for(i = 0; i < tripletMatrix1.length; i++)
+			{
+				rowOfMatrix1 = tripletMatrix1[i][0];
+				columnOfMatrix1 = tripletMatrix1[i][1];
+	
+				int index = indexAlreadyPresent(tripletMatrix2, rowOfMatrix1, columnOfMatrix1);
+				
+				if(index >= 0)
+				{
+					resultantMatrix[resultantMatrixRow][0] = rowOfMatrix1;
+					resultantMatrix[resultantMatrixRow][1] = columnOfMatrix1;
+					resultantMatrix[resultantMatrixRow][2] = tripletMatrix1[i][2] + tripletMatrix2[index][2];
+					resultantMatrixRow++;
+				}
+				else
+				{
+					resultantMatrix[resultantMatrixRow][0] = rowOfMatrix1;
+					resultantMatrix[resultantMatrixRow][1] = columnOfMatrix1;
+					resultantMatrix[resultantMatrixRow][2] = tripletMatrix1[i][2];
+					resultantMatrixRow++;
+				}
+			}
+			
+			for(i = 0; i < tripletMatrix2.length; i++)
+			{
+				int index = indexAlreadyPresent(resultantMatrix, tripletMatrix2[i][0], tripletMatrix2[i][1]);
+				if(index < 0)
+				{
+					
+					resultantMatrix[resultantMatrixRow][0] = tripletMatrix2[i][0];
+					resultantMatrix[resultantMatrixRow][1] = tripletMatrix2[i][1];
+					resultantMatrix[resultantMatrixRow][2] = tripletMatrix2[i][2];
+					resultantMatrixRow++;
+				}
+			}
+			
+		}
+		else
```

Comment:
There is no need for handling AssertionError. If you feel there is need then use assert keyword.